pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PulsarCompactorSubscription.java

```diff
@@ -106,5 +108,16 @@ public void markDeleteFailed(ManagedLedgerException exception, Object ctx) {
         }
     }
 
+    CompletableFuture<Void> cleanCompactedLedger() {
```

Comment:
We have `TopicCompactionService` now, shall we add an new API to the `TopicCompactionService`?

Reply:
This PR is just to fix this problem, it needs to be cherry-picked to release branch, I think we can separately propose a PIP extending TopicCompactionService on later version.

Reply:
Ok, sure.